lighthouse-core/test/chromium-web-tests/test-page-devtools.sh

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bash
```

Comment:
You should be able to reuse most of `run-web-tests.sh` given these changes:

1. remove the parameter `http/tests/devtools/lighthouse` in run_web_tests.py invoke
2. pass `http/tests/devtools/lighthouse` to `run-web-tests.sh` in `test-locally.sh`
3. write layout test in `http/tests/devtools/lighthouse`
4. new script `test-page-devtools.sh` copies file to `$DEVTOOLS_PATH/http/tests/devtools/test_lighthouse` folder, calls `run-web-tests.sh http/tests/devtools/test_lighthouse` (plus any other needed options, like `--no-retry-failures`), then grabs the LHR from output folder

Reply:
I think I can create some version of this but we will probably need to put `lighthouse-run-dt.js` in a `.gitignore`. `lighthouse-run-dt.js` needs to be created at runtime because the script inserts the desired url into the file.

Reply:
haha oh, right. you can keep it in the same place but with a placeholder value `const url ='__URL__'`, then during step 4) where you copy the script you also edit it, once it is already in the devtools folder.

Reply:
When we execute `run-web-tests.sh` it will roll devtools in which will override the `lighthouse-run-dt.js` file if we copy it there before executing `run-web-tests.sh`.
https://github.com/GoogleChrome/lighthouse/blob/efe63fff0d0645a3eb738443b4c13f74bd27071e/lighthouse-core/test/chromium-web-tests/run-web-tests.sh#L32-L36

To avoid this, `lighthouse-run-dt.js` (with custom url inserted) must be put with the regular web tests in `third-party/chromium-webtests/webtests/http/tests/devtools/lighthouse` _before_ executing `run-web-test.sh` so it gets copied when rolling devtools.

The current script doesn't need to do this because it puts `lighthouse-run-dt.js` directly into the devtools folder _after_ rolling devtools but before actually running the tests.

Reply:
Alright, let's put it there then. We really don't want to separately maintain all the setup nastiness.

Reply:
Sorry, I should have mentioned how that was problematic too.

I don't see a way to use `run-web-test.sh` as-is without needing to separately maintain a temporary file popping up in `third-party/chromium-webtests/webtests/http/tests/devtools/lighthouse`. We could remove the file when the script exits, but the file might persist if the script is interrupted. This is where a `.gitignore` would be needed.

One other option is to split `run-web-tests.sh` into two parts, one part to roll devtools and one part to run web tests. We can manually copy the file into devtools in between running these scripts. This way we don't have a bunch of copied code and we can avoid the file shuffling in directories tracked by git.

Reply:
> One other option is to split run-web-tests.sh into two parts, one part to roll devtools and one part to run web tests. We can manually copy the file into devtools in between running these scripts. This way we don't have a bunch of copied code and we can avoid the file shuffling in directories tracked by git.

This sounds great 👍 I also share Connor's concern about maintaining this between them and this solution sounds like better readability for the steps involved in run-web-tests.sh anyway 🎉  I'll also add that some of this breakup might be necessary for efficient large scale usage of this method, so separating the setup from the running is triple useful.

One other suggestion that might help a bit @adamraine is that if there are any sections here duplicated elsewhere (even after the split), super clear comment demarkation about what's shared and what's specific to this hackiness would be awesome!